### PR TITLE
tests: bsim: bluetooth: host: gatt: general: improve conn ref handling

### PR DESCRIPTION
**Pull request by kapi-no**
Improved the connection reference handling in the BabbleSim test project in the Bluetooth Host category.

### CODE REVIEW

File: tests/bsim/bluetooth/host/gatt/general/src/gatt_client_test.c [hunk 1]
```diff
@@ -54,7 +54,8 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 
 	bt_conn_unref(g_conn);
 
-	g_conn = NULL;
+	__ASSERT_NO_MSG(g_conn == conn);
+
```
Comment (Thalley) [inline tests/bsim/bluetooth/host/gatt/general/src/gatt_client_test.c:58]: Should this have been at line 39?

Setting `g_conn = NULL` after an unref seems like the right thing to do.

Comment (kapi-no) [reply tests/bsim/bluetooth/host/gatt/general/src/gatt_client_test.c:58]: yes, sorry

Comment (kapi-no) [reply tests/bsim/bluetooth/host/gatt/general/src/gatt_client_test.c:58]: Thank you for catching this!

Comment (kapi-no) [reply tests/bsim/bluetooth/host/gatt/general/src/gatt_client_test.c]: this is fixed now

**Review by Thalley — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Pull request merged**